frontend/src/metabase/query_builder/actions/navigation.js

```diff
@@ -60,6 +60,7 @@ export const popState = createThunkAction(
         const shouldRefreshUrl = location.state.card.dataset;
         await dispatch(setCardAndRun(location.state.card, shouldRefreshUrl));
         await dispatch(setCurrentState(location.state));
+        await dispatch(resetUIControls());
```

Comment:
Could we modify `export const uiControls = handleActions` reducer instead and handle `RESET_QB` action?

Reply:
I'll need to check that RESET_QB is dispatched, but if so then sure!

Reply:
In the scenario above, the `RESET_QB` action isn't actually dispatched (and neither is Initialize QB). I think that's mainly the issue here. I can try dispatching `RESET_QB` though and seeing if that helps

Reply:
actually, dispatching RESET_QB causes us to just see a loading screen :/

Reply:
> we aren't reseting UI controls in all cases when the QB is initialized

`initializeQB` [triggers `resetQB`](https://github.com/metabase/metabase/blob/1efa6c6f000cb4bd38065563176c35ea0027c865/frontend/src/metabase/query_builder/actions/core/initializeQB.ts#L244), but it looks like `initializeQB` purposely [isn't dispatched when navigating back](https://github.com/metabase/metabase/blob/cf1d27c9fea3d64db86a1a0a6f15ecb0be92c71c/frontend/src/metabase/query_builder/actions/navigation.js#L97-L113).

BTW, the custom [back-navigation logic](https://github.com/metabase/metabase/blob/cf1d27c9fea3d64db86a1a0a6f15ecb0be92c71c/frontend/src/metabase/query_builder/actions/navigation.js#L35-L-79) is very buggy:
- #33709
- #34514
- #33708

It looks like dispatching `resetUIControls` is the cheapest effective measure we can take in this case without reworking the entire thing. I tested it manually a bit and I haven't found any regressions due to this change.